pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -212,12 +212,12 @@ func (r *RemoteService) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	u.RawQuery = req.URL.RawQuery
 
 	proto := req.Header.Get("X-Forwarded-Proto")
-	if proto != "" {
+	req.URL.Scheme = "https"
+	// Certain reverse proxies may use the non-standard wss for websockets; allow https to remain the default.
+	if proto != "" && proto != "wss" {
```

Comment:
If proto == 'proto' and req.TLS == nil then the scheme will be "http".  We want to only consider the req.TLS parameter if proto isn't set.  This is because TLS is typically terminated early to https connections will req.TLS == nil.

Reply:
I'm not totally sure my comment it correct, but I think the logic is off.

Reply:
@ibuildthecloud Thanks for the review. I think you're right. I've updated this with a specific override to transliterate `wss` to `https` - it's not as clean but I think given the issues you raise and the intent of this logic, that's the only way to do what we're attempting here.